db/dcrpg/pgblockchain.go

```diff
@@ -3583,6 +3583,9 @@ func (pgb *ChainDB) StoreBlock(msgBlock *wire.MsgBlock, isValid, isMainchain,
 			return
 		}
 
+		// Signal updates to any subscribed heightClients.
+		pgb.SignalHeight(dbBlock.Height)
```

Comment:
One way to avoid the block summary sql error during startup sync might be to put this in the `if !pgb.InBatchSync {` block below.  I'd have to look to see how this was handled in the sqlite days.

Reply:
Actually, looking at the error log I posted, it's not every block.

Looking at `(*appContext).StatusNtfnHandler`, I think what we're seeing is a inherent race with `c.DataSource.GetBestBlockSummary()`.  It first gets the best block height from the meta table, and then attempts to retrieve block summary data for a block at that height (from the blocks and stats table).  But in ChainDB.StoreBlock, `InsertBlockStats` (and `SignalHeight`) are called after the meta table is updated.  So, I think SignalHeight needs to be after those two things, and SignalHeight should be after all DB updates.